qutebrowser-consult.el

```diff
@@ -0,0 +1,238 @@
+  "Sources used by `qutebrowser-launcher' and family.")
+
+(defun qutebrowser-consult--shorten-source-name (source)
+  "Return SOURCE with shortened name.
+
+'Qutebrowser' is stripped from the front and the remaining first letter is capitalized."
+  (let ((new-source (seq-copy source))
+	(new-name (string-remove-prefix "Qutebrowser " (plist-get source :name))))
+    (plist-put new-source :name (capitalize new-name))))
+
+(defun qutebrowser-consult--suppress-action (source)
+  "Return SOURCE with no action."
+  (let ((new-source (seq-copy source)))
+    (plist-put new-source :action nil)))
+
+;; NOTE does this still need to exist?
+(defun qutebrowser-consult-select-url (&optional initial default)
+  "Dynamically select a URL, buffer, or command using consult.
+INITIAL sets the initial input in the minibuffer."
+  (let ((consult-async-min-input 0)
+        (consult-async-split-style nil)
+	(sources (list qutebrowser-consult--command-source
+		       qutebrowser-consult--exwm-buffer-source
+		       qutebrowser-consult--bookmark-source
+		       qutebrowser-consult--history-source)))
+    (consult--multi
+     (mapcar #'qutebrowser-consult--shorten-source-name
+	     (mapcar #'qutebrowser-consult--suppress-action sources))
+     :prompt (if default
+                 (format "Select (default %s): " default)
+               "Select: ")
+     :default default
+     :sort nil
+     :initial initial
+     :require-match nil)))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher (&optional initial target)
+  "Select a URL to open in Qutebrowser using Consult.
+
+Set initial completion input to INITIAL. Open the URL in TARGET or the
+default target if nil.
+
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (let* ((consult-async-split-style nil)
+	 (qutebrowser-default-open-target (or target qutebrowser-default-open-target))
+	 (selected (consult--multi
+		    (mapcar #'qutebrowser-consult--shorten-source-name
+			    (mapcar #'eval qutebrowser-consult-launcher-sources))
+		    :initial initial
+		    :sort nil)))
+    (unless (plist-get (cdr selected) :match)
+      (qutebrowser (car selected)))))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-tab (&optional initial)
+  "Select a URL to open in a new tab using Consult.
+
+Set initial completion input to INITIAL.
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (qutebrowser-consult-launcher initial 'tab))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-window (&optional initial)
+  "Select a URL to open in a new window using Consult.
+
+Set initial completion input to INITIAL.
+Modify `qutebrowser-consult-launcher-sources' to change which sources
+are included."
+  (interactive)
+  (qutebrowser-consult-launcher initial 'window))
+
+;;;###autoload
+(defun qutebrowser-consult-launcher-private (&optional initial)
+  "Select a URL to open in a private window using Consult.
```

Comment:
I don't think we need to duplicate all of these launcher functions.

I think it would be better to add a defcustom in the `qutebrowser.el` file to select which launcher backend should be used for the various launcher functions. Similar to how it is done for `qutebrowser-command-backend`.

The variable could be called something like `qutebrowser-select-url-backend` or `qutebrowser-launcher-backend` depending on how it is implemented.